src/rgw/rgw_rest.cc

```diff
@@ -781,7 +792,18 @@ int recv_body(struct req_state* const s,
               const size_t max)
 {
   try {
-    return RESTFUL_IO(s)->recv_body(buf, max);
+    int len = RESTFUL_IO(s)->recv_body(buf, max);
```

Comment:
only the call to `recv_body()` should be covered by the try-catch. that means moving `int len` out of the try block